src/miner_hbbft_sidecar.erl

```diff
@@ -0,0 +1,276 @@
+            lager:warning("response for unknown ref"),
+            {noreply, State};
+        #validation{from = From, pid = Pid, txn = Txn, monitor = MRef} ->
+            Reply =
+                case Res of
+                    ok ->
+                        case blockchain_txn:absorb(Txn, Chain) of
+                            ok ->
+                                catch libp2p_group_relcast:handle_command(Group, Txn),
+                                ok;
+                            Error ->
+                                lager:warning("speculative absorb failed for ~p, error: ~p", [Txn, Error]),
+                                Error
+                        end;
+                    deadline ->
+                        erlang:exit(Pid, kill),
+                        write_txn("timed out", Height, Txn),
+                        {error, validation_deadline};
+                    {error, Error} ->
+                        write_txn("failed", Height, Txn),
+                        lager:warning("is_valid failed for ~p, error: ~p", [Txn, Error]),
+                        Error
+                end,
+            erlang:demonitor(MRef, [flush]),
+            gen_server:reply(From, Reply),
+            Validations1 = maps:remove(Ref, Validations),
+            {noreply, maybe_start_validation(State#state{validations = Validations1})}
+    end;
+handle_info({'DOWN', Ref, process, _Pid, Reason}, #state{validations = Validations} = State) ->
+    case maps:to_list(maps:filter(
+                        fun(_K, #validation{monitor = MRef}) when Ref == MRef -> true;
+                           (_, _) -> false end,
+                        Validations)) of
+        [{Attempt, #validation{from = From}}] ->
+            gen_server:reply(From, {error, validation_crashed}),
+            Validations1 = maps:remove(Attempt, Validations),
+            {noreply, maybe_start_validation(State#state{validations = Validations1})};
+        _ ->
+            lager:warning("DOWN msg for unknown ref. pid = ~p reason = ", [_Pid, Reason]),
+            {noreply, State}
+   end;
+handle_info(chain_check, State) ->
+    {ok, State1} = init(State),
+    {noreply, State1};
+handle_info(_Info, State) ->
+    lager:warning("unexpected message ~p", [_Info]),
+    {noreply, State}.
+
+terminate(_Reason, #state{validations = Validations}) ->
+    maps:map(
+      fun(_K, #validation{from = From, pid = Pid}) ->
+              gen_server:reply(From, {error, exiting}),
+              erlang:exit(Pid, kill)
+      end, Validations),
+    ok.
+
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+filter_txn_buffer(Buf, Chain) ->
+    lists:filter(fun(BinTxn) ->
+                         Txn = blockchain_txn:deserialize(BinTxn),
+                         case blockchain_txn:is_valid(Txn, Chain) of
```

Comment:
if these are "slow" transactions, can we assume their absorbs will fail because of the changes we made in core and avoid calling the validate?

Reply:
And if the above is true, maybe we can go back to doing filtering inline in the hbbft process so we avoid potentially losing transactions because of a crash because relcast will save our buffer again